Feat/get dispatch detail#47

myoungjin-kim opened the pull request:
<!-- PR 예시 : Feat: Issues Template 및 Pull Request Template 추가 --!>
<!-- 라벨과 리뷰어를 등록해주세요. -->

## 작업 대상
<!-- 작업한 대상을 자세히 설명해주세요. -->

차량 관제에서 특정 배차의 상세 정보를 확인하는 기능입니다.

## 📄 작업 내용
<!-- 작업한 내용을 간단히 요약해주세요. -->

입력 받은 배차 번호에 해당하는
배차 정보와 배차 상세 정보를 조회하는 기능을 구현합니다.

## 🙋🏻 주의 사항
<!-- 리뷰어를 위해 복잡하거나 중요한 코드를 명시해주세요. -->

## 스크린샷
<!-- 필요한 경우 이해를 돕기위해 스크린샷을 올려주세요. -->

## 📎 관련 이슈
<!-- merge 시 close할 issue 번호를 입력해주세요. -->

<!-- closed #번호 --> 
closed #47
## 레퍼런스
<!-- 작업 시 참고했던 문건의 URL을 남겨주세요 -->

myoungjin-kim committed:
feat: 엔티티 수정

VehicleType enum으로 변경
VehicleDetail.vehicleName -> VehicleDetail.vehicleTon
DispatchDetailStatus 추가
DispatchDetail.ett, DispatchDetail.isWaiting 추가

myoungjin-kim committed:
feat: 차량 관제 배차 상세 확인 Controller, Response 추가

myoungjin-kim committed:
refactor: TransportOrder, Vehicle, VehicleDetail 엔티티 수정

myoungjin-kim committed:
fix: 코드 누락

myoungjin-kim committed:
refactor: DispatchDetailResponse 자료형 수정

myoungjin-kim committed:
feat: DispatchDetailService 추가

Sm, DispatchRepository, DispatchErrorCode,
DeliveryDestinationRepository, CenterRepository
수정

myoungjin-kim committed:
feat: SwaggerDispatchDetailController 추가

myoungjin-kim committed:
Merge remote-tracking branch 'origin/dev' into feat/get-dispatch-detail#47

myoungjin-kim committed:
Merge remote-tracking branch 'origin/dev' into feat/get-dispatch-detail#47

myoungjin-kim committed:
fix: isWaiting -> isResting 변수 명 수정, isResting에 따른 배차 상세 상테 반환값 로직 추가

sinsj5858 commented on src/main/java/com/team2/finalproject/domain/dispatchdetail/model/dto/response/DispatchDetailResponse.java:
```diff
@@ -0,0 +1,131 @@
+        @Schema(example = "127.1116", description = "경도")
+        private Double lon;
+    }
+
```
저희 출발지도 예정 출발시간이 필요하지 않을까 싶습니다.
전체 상차시작시간과 상차종료한시간(운행시작시간) 이있는데 후자를 추가해야하지 않을까 싶습니다.
dispatchDetail리스트의 첫번째 값을 출발지라고 생각하고 조회하면 될듯 합니다. (배차 확정과 같이 회의 해야할듯?)

sinsj5858 commented on src/main/java/com/team2/finalproject/domain/dispatchdetail/service/DispatchDetailService.java:
```diff
@@ -0,0 +1,113 @@
+    }
+
+    private DispatchDetailResponse.DispatchDetail getDispatchDetailList(DispatchDetail dispatchDetail) {
+
```
DispatchDetailResponse.DispatchDetail 내부 클래스에서 DispatchDetail이나 DispatchDetail와 TransportOrder를 받아 반환하는 코드를 사용하면 service코드가 아닌 dto클래스 내부에서 객체를 생성하고 반환할 수 있을것  같습니다

myoungjin-kim replied:
수정 했습니다!

sinsj5858 reviewed (approved):
고생하셨습니다

qudwns017 reviewed (approved):
고생하셨습니다!

myoungjin-kim committed:
refactor: DispatchDetailResponse 내부 객체 생성

DispatchDetailResponse.StartStopover,

DispatchDetailResponse.DispatchDetail 생성

DispatchDetailService -> DispatchDetailResponse

myoungjin-kim committed:
feat: DispatchDetailResponse departureTime(운송 시작 시간) 추가
